src/mouse-joystick-source.js

```diff
@@ -0,0 +1,179 @@
+
+    /**
+     * Handle global mouse movement (even when not over canvas)
+     * @param {MouseEvent} event - The mouse event
+     */
+    handleGlobalMouseMove(event) {
+        const rect = this.canvas.getBoundingClientRect();
+        const x = event.clientX - rect.left;
+        const y = event.clientY - rect.top;
+
+        // Normalize to 0-1 range
+        this.mouseX = x / rect.width;
+        this.mouseY = y / rect.height;
+
+        // Clamp values to 0-1 range
+        this.mouseX = Math.max(0, Math.min(1, this.mouseX));
+        this.mouseY = Math.max(0, Math.min(1, this.mouseY));
+    }
+
+    /**
+     * Handle mouse button press
+     * @param {MouseEvent} event - The mouse event
+     */
+    handleMouseDown(event) {
+        if (!this.isActive || !this.via) return;
+
+        // Only handle left mouse button (button 0)
+        if (event.button === 0) {
+            // Set fire button 1 pressed (PB4)
+            this.via.setJoystickButton(0, true);
+            event.preventDefault();
+        }
+    }
+
+    /**
+     * Handle mouse button release
+     * @param {MouseEvent} event - The mouse event
+     */
+    handleMouseUp(event) {
+        if (!this.via) return;
+
+        // Only handle left mouse button (button 0)
+        if (event.button === 0) {
+            // Release fire button 1 (PB4)
+            this.via.setJoystickButton(0, false);
+            event.preventDefault();
+        }
+    }
+
+    /**
+     * Set the VIA reference for button handling
+     * @param {object} via - The system VIA
+     */
+    setVia(via) {
+        this.via = via;
+    }
+
+    /**
+     * Get analog value from mouse position for the specified channel
+     * @param {number} channel - The ADC channel (0-3)
+     * @returns {number} A value between 0 and 0xffff
+     */
+    getValue(channel) {
+        let value;
+
+        // Use mouse position when enabled (always active when assigned to a channel)
+        {
+            switch (channel) {
+                case 0:
+                    // X axis for joystick 1
+                    // Convert from [0,1] to [0,0xffff]
+                    value = Math.floor(this.mouseX * 0xffff);
+                    break;
+                case 1:
+                    // Y axis for joystick 1
+                    // Convert from [0,1] to [0,0xffff]
+                    value = Math.floor(this.mouseY * 0xffff);
+                    break;
+                case 2:
+                    // X axis for joystick 2 (not used for mouse)
+                    value = 0x8000;
+                    break;
+                case 3:
+                    // Y axis for joystick 2 (not used for mouse)
+                    value = 0x8000;
+                    break;
+                default:
+                    value = 0x8000;
+                    break;
+            }
+        }
+
+        return value;
+    }
+
+    /**
+     * Clean up event listeners when source is no longer needed
+     */
+    dispose() {
```

Comment:
What calls this function? I'm happy to leave it in, but wondered if it needs to be wired up anywhere.